src/webapp/components/replicate/ReplicateUserFromTemplate.tsx

```diff
@@ -0,0 +1,368 @@
+                                passwordTemplate: `${randomPasswordBase}_$index`,
+                            }}
+                            validate={values => {
+                                const errors = formValidator(values, existingUsernames);
+                                return errors;
+                            }}
+                            autocomplete="off"
+                            render={({ handleSubmit }) => (
+                                <>
+                                    <FormSpy
+                                        onChange={(state: FormState<FormValues>) => {
+                                            requestAnimationFrame(() => {
+                                                setValidationError(!_.isEmpty(state.errors));
+                                            });
+                                        }}
+                                    />
+
+                                    <form id="replicate-form" onSubmit={handleSubmit}>
+                                        <DialogContent>
+                                            <RenderFormField name="replicateCount" label="Number of users to create" />
+                                            <RenderFormField
+                                                name="usernameTemplate"
+                                                label="Username template"
+                                                helpText={i18n.t(
+                                                    "Using $index is optional if only one replica is created. Multiple $index placeholders can be used anywhere in the fields."
+                                                )}
+                                            />
+                                            <RenderFormField
+                                                name="passwordTemplate"
+                                                label="Password template"
+                                                helpText={i18n.t(
+                                                    "Please ensure to copy the password and adhere to security guidelines"
+                                                )}
+                                            />
+                                        </DialogContent>
+                                    </form>
+                                </>
+                            )}
+                        />
+                    </DialogContent>
+
+                    <DialogActions>
+                        <Button onClick={onRequestClose}>{i18n.t("Cancel")}</Button>
+                        <Button disabled={hasValidationErrors} type="submit" form="replicate-form" color="primary">
+                            {i18n.t("Replicate")}
+                        </Button>
+                    </DialogActions>
+                </Dialog>
+            )}
+        </>
+    );
+};
+
```

Comment:
These validations are business rules validations and should be realized in the user domain entity.

To validate you can to try to create a domain entity, call internally to validate a return, for example an Either with the entity or the errors.

You can see these examples:
* https://github.com/EyeSeeTea/malaria-threats-map/blob/c69d08e377d83d56f2d08da54899616b97f5492c/src/domain/entities/Feedback.ts#L36
* https://github.com/EyeSeeTea/coding-dojo/blob/kata/clean-architecture-solution/src/domain/entities/Product.ts